submissions/asaMitaka/FriendsApp/script.js

```diff
@@ -0,0 +1,161 @@
+const url = 'https://randomuser.me/api/?results=12';
+const content = document.querySelector('.articleBlock');
+const ageForm = document.querySelector('#ageForm');
+const genderForm = document.querySelector('#genderForm');
+const search = document.querySelector('#search');
+let friends = [];
+let friendsCopy = [];
+
+const data = (url) => fetch(url)
+    .then(handleErrors)
+    .then(res => res.json())
+    .then(res =>  {
+        friends = [...res.results];
+        init();
+    });
+
+function handleErrors(res) {
+    if (!res.ok) {
+      throw Error(res.statusText);
+    }
+    return res;
+}
+
+const filterState = {
+    search: null, 
+    gender: null,
+    sort: null,
+    reset() {
+        this.search = null;
+        this.gender = null;
+        this.sort = null;
+    }
+}
+
+function filterBy(id, usersToFilter) {
```

Comment:
I believe the only id you have there is somehow connected with gender, so `filterBy` is actually `filterByGender`.